src/components/v2/values/chartValuesDiff/ChartValuesView.tsx

```diff
@@ -0,0 +1,1128 @@
+                (e) => e.value.toString() === commonState.installedConfig.environmentId.toString(),
+            )
+            dispatch({
+                type: 'multipleOptions',
+                payload: {
+                    selectedProject: project,
+                    selectedEnvironment: environment,
+                },
+            })
+        }
+    }, [commonState.installedConfig, commonState.environments, commonState.projects])
+
+    useEffect(() => {
+        if (
+            commonState.activeTab === 'manifest' &&
+            commonState.installedConfig &&
+            (!commonState.generatedManifest ||
+                (commonState.generatedManifest &&
+                    (hasChartChanged() || commonState.chartValues.id !== commonState.installedConfig.id)))
+        ) {
+            if (isDeployChartView) {
+                getGeneratedHelManifest(
+                    commonState.selectedEnvironment.value,
+                    commonState.selectedEnvironment.clusterId || commonState.installedConfig.clusterId,
+                    commonState.selectedEnvironment.namespace,
+                    appName,
+                    commonState.chartValues?.appStoreVersionId ||
+                        commonState.chartValues?.id ||
+                        commonState.installedConfig.appStoreVersion,
+                    commonState.installedConfig.valuesYaml,
+                    dispatch,
+                )
+            } else {
+                getGeneratedHelManifest(
+                    commonState.installedConfig.environmentId,
+                    commonState.installedConfig.clusterId,
+                    commonState.installedConfig.namespace,
+                    commonState.installedConfig.appName,
+                    commonState.chartValues?.appStoreVersionId ||
+                        commonState.chartValues?.id ||
+                        commonState.installedConfig.appStoreVersion,
+                    isExternalApp
+                        ? commonState.releaseInfo.mergedValues
+                        : commonState.installedConfig.valuesOverrideYaml,
+                    dispatch,
+                )
+            }
+        }
+    }, [commonState.activeTab, commonState.installedConfig])
+
+    useEffect(() => {
+        if (chartValuesList.length > 0 || commonState.deploymentHistoryArr.length > 0) {
+            const isVersionAvailableForDiff =
+                chartValuesList.some((_chartValues) => _chartValues.kind === 'DEPLOYED') ||
+                commonState.deploymentHistoryArr.length > 0
+
+            dispatch({ type: 'isComparisonAvailable', payload: isVersionAvailableForDiff })
+        }
+    }, [chartValuesList, commonState.deploymentHistoryArr])
+
+    const initData = async (_installedAppInfo: InstalledAppInfo, _releaseInfo: ReleaseInfo) => {
+        try {
+            const { result } = await getChartVersionDetails2(_installedAppInfo.installedAppVersionId)
```

Comment:
can we rename getChartVersionDetails2?

Reply:
Done. Renamed to `getChartVersionDetailsV2` indicating that it's a v2 of the chart details API.